awstesting/behavior_tests/sampleresetxmlprotocolservice/behav_test.go

```diff
@@ -0,0 +1,7351 @@
+			return false
+		}
+	}
+	return true
+}
+
+func assertRequestBodyEqualsBytes(t *testing.T, req *request.Request, val string) bool {
+	var bytesReqBody []byte
+	bytesExpect, err := base64.StdEncoding.DecodeString(val)
+
+	if err != nil {
+		t.Errorf("expect no error, got %v", err)
+	}
+
+	if req.HTTPRequest.Body != nil {
+		bytesReqBody, err = ioutil.ReadAll(req.HTTPRequest.Body)
+		if err != nil {
+			t.Errorf("expect no error, got %v", err)
+		}
+	}
+
+	req.HTTPRequest.Body = ioutil.NopCloser(bytes.NewBuffer(bytesReqBody))
+
+	return bytes.Compare(bytesReqBody, bytesExpect) == 0
+}
+
+func assertRequestBodyEqualsJson(t *testing.T, req *request.Request, val string) bool {
+	var bytesReqBody []byte
+	var err error
+	if req.HTTPRequest.Body != nil {
+		bytesReqBody, err = ioutil.ReadAll(req.HTTPRequest.Body)
+		if err != nil {
+			t.Errorf("expect no error, got %v", err)
+		}
+	}
+
+	req.HTTPRequest.Body = ioutil.NopCloser(bytes.NewBuffer(bytesReqBody))
+
+	return awstesting.AssertJSON(t, val, util.Trim(string(bytesReqBody)))
+}
+
+func assertRequestBodyMatchesXml(t *testing.T, req *request.Request, val string, container interface{}) bool {
+	r := req.HTTPRequest
+
+	if r.Body == nil {
+		t.Errorf("expect body not to be nil")
+	}
+	body := util.SortXML(r.Body)
+
+	return awstesting.AssertXML(t, val, util.Trim(string(body)), container)
+}
+
+func assertRequestBodyEqualsString(t *testing.T, req *request.Request, val string) bool {
+	var bytesReqBody []byte
+	var err error
+	if req.HTTPRequest.Body != nil {
+		bytesReqBody, err = ioutil.ReadAll(req.HTTPRequest.Body)
+		if err != nil {
+			t.Errorf("expect no error, got %v", err)
+		}
+	}
+
+	req.HTTPRequest.Body = ioutil.NopCloser(bytes.NewBuffer(bytesReqBody))
+	stringReqBody := string(bytesReqBody)
+
+	return stringReqBody == val
+}
+
+func assertRequestIdEquals(t *testing.T, req *request.Request, val string) bool {
+	return req.RequestID == val
+}
+
+func assertResponseDataEquals(t *testing.T, response interface{}, expectResponse interface{}) bool {
+	if response == nil || expectResponse == nil {
+		return response == expectResponse
+	}
+	return cmp.Equal(expectResponse, response)
+}
+
+func assertResponseErrorIsKindOf(t *testing.T, err error, val string) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return awsErr.Code() == val
+	}
+	return true
+}
+
+func assertResponseErrorMessageEquals(t *testing.T, err error, val string) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return awsErr.Message() == val
+	}
+	return true
+}
+
+func assertResponseErrorDataEquals(t *testing.T, err error, val map[string]interface{}) {
+	if testing.Short() {
+		t.Skip("skipping responseErrorDataEquals assertion")
+	}
+}
+
+func assertResponseErrorRequestIdEquals(t *testing.T, err error, val string) bool {
+	if reqErr, ok := err.(awserr.RequestFailure); ok {
+		return reqErr.RequestID() == val
+	}
+	return true
+}
+
+//Can build empty PUT requests
+func BehavTest_00(t *testing.T) {
```

Comment:
These should be in the form of `TestBehavior_xx`. Go requires unit tests functions to start with `Test`